lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -166,35 +165,37 @@ public function exec(string $statement) : int
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
-        return $this->conn->insert_id;
+        if ($this->conn->insert_id === 0) {
```

Comment:
Let's assign the value of `$this->conn->insert_id` to a variable and use it in all cases for better atomicity.

Reply:
Fixed!